common/content/finder.js

```diff
@@ -220,23 +223,19 @@ var Finder = Module("finder", {
             this.find(pattern);
         }
 
-        this._lastSearchPattern = pattern;
-
         // TODO: move to find() when reverse incremental searching is kludged in
         // need to find again for reverse searching
         if (this._backwards)
             this.findAgain(true);
 
         if (options["hlsearch"])
-            this.highlight(findbar._findField.value);
+            this.highlight();
     },
 
     /**
      * Highlights all occurances of <b>str</b> in the buffer.
-     *
-     * @param {string} str The string to highlight.
      */
-    highlight: function (str) {
+    highlight: function () {
```

Comment:
Didn't realize this previously, does this change mean that someone else can't use `findbar.highlight('foo')` to highlight `foo`?